src/filesystem.rs

```diff
@@ -776,6 +818,51 @@ fn compare_symlink(
     }
 }
 
+#[derive(Debug, PartialEq)]
+pub enum CopyComparison {
+    Identical,
+    OnlySourceExists,
+    OnlyTargetExists,
+    Changed,
+    TargetNotRegularFileOrDirectory,
```

Comment:
I think this should be `TargetNotRegularFile` since directories fail the check at line 862 as well.

Reply:
Yeah, you're right about that. I'll change it.